kahuna/public/js/upload/manager.js

```diff
@@ -30,7 +30,7 @@ upload.factory('uploadManager',
 
         // once all `jobItems` in a job are complete, remove it
         // TODO: potentially move these to a `completeJobs` `Set`
-        $q.all(promises).then(() => jobs.delete(job));
+        $q.all(promises).finally(() => jobs.delete(job));
```

Comment:
This is the fix for uploading a bad image, navigating around then navigate back to /upload.